### ISSUE TITLE
Hide checkboxes on singleSelection: true

### ISSUE THREAD

**Issue by scoopz** [NileshPatel17/ng-multiselect-dropdown#270]
When the singleSelection is true, there is no need of showing checkboxes in my opinion because anyways the user can select only one item. so if that prop is true, it should work just like a normal combobox with the lovely typeahead. also, that selected item tag preview is also not appropriate at that point because there will be only one selection so we can show just the selected value instead of as a tag. I really enjoyed this plugin and loved your work @NileshPatel17 

Thanks,

ref: **selected item tag preview**
[
![Capture](https://user-images.githubusercontent.com/26799723/94984696-63913680-056c-11eb-95fb-4efa230ba5e3.PNG)
](url)

**Comment by exsc185** [comment]
Hi,
Any idea about this? Also would be great if we can even show radio button circle instead of checkbox.

**Comment by scoopz** [comment]
@exsc185 In my opinion, we don't even need to have radio buttons. because whenever we choose an item from the dropdown list if the singleSelection is true, the dropdown should hide automatically revealing the selected item right because there are no more choices to make. if the user changed his mind, he can open the dropdown again and choose whatever they want. **So basically, it should work as a raw HTML `<select>` tag with added typeahead search functionality when the singleSelection is true.**

**Comment by exsc185** [comment]
@scoopz , yes makes sense.
But just from look and feel perspective radio button or just values without checkbox would look better.

**Comment by scoopz** [comment]
@exsc185 yeah, that's what I mentioned. we don't need the checkboxs nor the radio buttons. Just the dropdown with text values would work for single selection.

**Comment by exsc185** [comment]
@scoopz 100%

**Comment by NileshPatel17** [comment]
@exsc185 @scoopz This change will be coming up soon.

**Comment by kinaski** [comment]
Hi folks, any new solution about this: "Hide checkboxes on singleSelection"?
Thanks in In advance  @NileshPatel17 ?

**Comment by sireeshap** [comment]
hi @NileshPatel17  Thanks for nice library. any update above issue. i am also looking for this enhancement.

**Comment by himanshumittal90** [comment]
you can add below css by adding singleSelection class in your element

.singleSelection .dropdown-list input[type="checkbox"]+div:before {
  display: none;
}

.singleSelection input[type="checkbox"]+div:after {
  display: none;
}

.singleSelection input[type="checkbox"]+div {
  padding: 6px 10px !important;
}

.singleSelection input[type="checkbox"]:checked+div {
  background-color: #337ab7 !important;
  color: #fff !important;
}

**Comment by ShivamTiwari0010** [comment]
::ng-deep .dropdown-list ul li.multiselect-item-checkbox div:before {
  display: none;
}
::ng-deep .dropdown-list ul li.multiselect-item-checkbox div:after {
  display: none;
}
::ng-deep .dropdown-list ul li.multiselect-item-checkbox div {
  padding-left: 0 !important;
}

**Comment by Osadha-Ganewaththa** [comment]
You can use Angular's ViewEncapsulation strategies or global styles as alternatives.

1️⃣ Use Global Styles 
Instead of ::ng-deep, move the CSS to styles.scss or styles.css

.first-class div:before,
.another-class div:before,
.third-class div:before {
    display: none;
}

.first-class div:after,
.another-class div:after,
.third-class div:after {
    display: none;
}

.first-class div,
.another-class div,
.third-class div {
    padding-left: 0 !important;
}
🔹 Works best for styles affecting multiple components globally.




2️⃣Use ViewEncapsulation with :host ::ng-deep (Scoped)

If you want to scope the styles only within a specific component, you can disable Angular's style encapsulation:


Step 1: Modify Component's TypeScript File
In your component (your.component.ts), set encapsulation to ViewEncapsulation.None:

import { Component, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'app-your-component',
  templateUrl: './your.component.html',
  styleUrls: ['./your.component.scss'],
  encapsulation: ViewEncapsulation.None,  // 👈 This allows global styling within this component
})
export class YourComponent { }

Step 2: Use Standard CSS in your.component.scss

.first-class div:before,
.another-class div:before,
.third-class div:before {
    display: none;
}

.first-class div:after,
.another-class div:after,
.third-class div:after {
    display: none;
}

.first-class div,
.another-class div,
.third-class div {
    padding-left: 0 !important;
}

🔹 Use this if you want styles to apply only inside this specific component.


3️⃣ Use :host Selector (Recommended for Components)

If styles should apply only inside a component, use :host instead of ::ng-deep:

:host .first-class div:before,
:host .another-class div:before,
:host .third-class div:before {
    display: none;
}

:host .first-class div:after,
:host .another-class div:after,
:host .third-class div:after {
    display: none;
}

:host .first-class div,
:host .another-class div,
:host .third-class div {
    padding-left: 0 !important;
}

🔹 Best if the styles are only for a single component and should not leak out.
